src/compiler/checker.ts

```diff
@@ -4738,6 +4739,14 @@ namespace ts {
             return checkTypeAssignableTo(source, target, /*errorNode*/ undefined);
         }
 
+        /**
+         * This is *not* a bi-directional relationship.
+         * If one needs to check both directions for comparability, use a second call to this function or 'checkTypeComparableTo'.

```

Comment:
See [this comment](https://github.com/Microsoft/TypeScript/pull/5517#discussion_r43935354)